### PR TITLE
UID resolved image scale url fix

### PR DESCRIPTION
**Pull request by petschki**
issue see https://github.com/plone/Products.CMFPlone/issues/2465

### CODE REVIEW

File: plone/app/textfield/transform.py [hunk 1]
```diff
@@ -67,3 +79,34 @@ def __call__(self, value, mimeType):
...
+        # get the original save time from the cached data dict
+        orig_time = getattr(cache_obj, cache._id).values()[0][0]
+
+        for ref_link in ol:
+            # only lookup image references
+            if IImage not in ref_link.to_interfaces_flattened:
+                continue
+            # XXX: not sure if it is a potential performance problem
+            # looking up the image object
+            if ref_link.to_object.modified() > orig_time:
```
Comment (jensens) [inline plone/app/textfield/transform.py]: I am not sure whats better, waking up the object or query the catalog. Without performance monitoring this is difficult to say.

**Comment by petschki** [comment]
@jensens thanks for your suggestion. I also wasn't satisfied with the object lookup from the beginning. I've refactored the referenced image lookup to pure catalog query without any dependency to p.a.linkintegrity anymore ... makes more sense for me

File: plone/app/textfield/transform.py [hunk 2]
```diff
@@ -67,3 +72,26 @@ def __call__(self, value, mimeType):
             # log the traceback of the original exception
             LOG.error("Transform exception", exc_info=True)
             raise TransformError('Error during transformation', e)
+
+    def check_referenced_images(self, target_mimetype, cache_obj):
+        # extract all image src uuids
+        uids = re.findall('src="[^/]*/resolve[uU]id/([^/"]*)', cache_obj.value)
```
Comment (jensens) [inline plone/app/textfield/transform.py]: Precompiling the regular expression saves some CPU cycles.

File: plone/app/textfield/transform.py [hunk 3]
```diff
@@ -67,3 +72,26 @@ def __call__(self, value, mimeType):
...
+        # referenced image scale urls get outdated if the images are modified.
+        # purging the transform cache updates those urls.
+        cache = Cache(cache_obj, context=self.context)
+        data = cache.getCache(target_mimetype)
+        if data is None:
+            # data is not cached
+            return
+
+        # get the original save time from the cached data dict
+        orig_time = getattr(cache_obj, cache._id).values()[0][0]
```
Comment (jensens) [inline plone/app/textfield/transform.py]: I would really like to see a catalog counter check (``portal_catalog.getCounter()``) here, which implies that we write the prior counter to some place, like a volatile attribute on the content object to be checked. If the count stays the same there is IMO no need for further checks.

Comment (petschki) [reply plone/app/textfield/transform.py]: I've implemented the catalog counter check and saved it to the `cache_obj` like PortalTransform.cache does

Comment (jensens) [reply plone/app/textfield/transform.py]: ok, I missed that one.

**Pull request merged**